src/StreamJsonRpc.Tests/JsonRpcTests.cs

```diff
@@ -1360,6 +1412,9 @@ private void ReinitializeRpcWithoutListening()
         this.serverRpc = new JsonRpc(this.serverMessageHandler, this.server);
         this.clientRpc = new JsonRpc(this.clientMessageHandler);
 
+        ((IJsonRpcInstanceContainer)this.clientMessageFormatter).Rpc = this.clientRpc;
+        ((IJsonRpcInstanceContainer)this.serverMessageFormatter).Rpc = this.serverRpc;
+
         this.serverRpc.TraceSource = new TraceSource("Server", SourceLevels.Verbose);
```

Comment:
We shouldn't be setting this in tests. The JsonRpc class itself should set this.

Reply:
Not sure I get this comment, do you mean the TraceSource? And what do you mean the JsonRpc should set it, should we create a method in JsonRpc to set his were we send the name and the SourceLevel?

---
In reply to: [303661727](https://github.com/milopezc/vs-streamjsonrpc/pull/1#discussion_r303661727) [](ancestors = 303661727)

Reply:
Sorry for lack of clarity here: I mean we should not be setting the Rpc property on the formatter. JsonRpc should assign that property when the formatter+handler is first passed to the JsonRpc ctor.

---
In reply to: [304099982](https://github.com/milopezc/vs-streamjsonrpc/pull/1#discussion_r304099982) [](ancestors = 304099982,303661727)

Reply:
Ok, I so when we pass the handler through the constructor the handler should contain the formatter right? But the handler is of type IJsonRpcMessageHandler, so I'd need to cast it to IJsonRpcInstanceContainer so that I can set the JsonRpc instance and not all formatters inherit from IJsonRpcInstanceContainer right? So how should I handle the case where the formatter doesn't have a rpc instance to set? Or maybe I'm misunderstanding again

---
In reply to: [304378610](https://github.com/milopezc/vs-streamjsonrpc/pull/1#discussion_r304378610) [](ancestors = 304378610,304099982,303661727)

Reply:
The JsonRpc ctor receives the handler, and the handler has a property on it to get the formatter. So you get the formatter that way, and use an `as` cast to try to cast the formatter to `IJsonRpcInstanceContainer`. If the cast succeeds, then you set the property.